File: JBTheatreToolsWin/Core.Tests/PassphraseTests.cs
using Xunit;

namespace JBTheatreTools.Tests;

/// <summary>
/// Verifies the passphrase reduction rule (case-insensitive; spaces &amp; punctuation ignored). Uses only
/// SYNTHETIC inputs — the real suite phrases are secrets and must never appear in this public repo. The
/// server applies the same rule to its private allow-list.
/// </summary>
public class PassphraseTests
{
    [Theory]
    [InlineData("HELLO", "hello")]
    [InlineData("HeLLo", "hello")]
    [InlineData("hello world", "helloworld")]
    [InlineData("  hello,  world!  ", "helloworld")]
    [InlineData("don't panic", "dontpanic")]
    [InlineData("a-b_c.d", "abcd")]
    [InlineData("Studio 54", "studio54")]
    [InlineData("!!! ,.- ", "")]
    public void NormalizesConsistently(string input, string expected)
        => Assert.Equal(expected, Passphrase.Normalize(input));

    [Fact]
    public void CaseAndSpacingVariantsCollapseToOneToken()
    {
        string[] variants = { "Foo Bar Baz", "foo bar baz", "FOOBARBAZ", "  foo,bar.baz  ", "Foo-Bar-Baz" };
        var normalized = new HashSet<string>();
        foreach (var v in variants) normalized.Add(Passphrase.Normalize(v));
        Assert.Single(normalized);
        Assert.Contains("foobarbaz", normalized);
    }
}

File: JBTheatreTools/Tests/JBTheatreToolsTests/PassphraseTests.swift
import XCTest
@testable import JBTheatreTools

/// Verifies the passphrase reduction rule (case-insensitive; spaces & punctuation ignored). Uses only
/// SYNTHETIC inputs — the real suite phrases are secrets and must never appear in this public repo. The
/// server applies the same rule to its private allow-list.
final class PassphraseTests: XCTestCase {
    func testLowercases() {
        XCTAssertEqual(Passphrase.normalize("HELLO"), "hello")
        XCTAssertEqual(Passphrase.normalize("HeLLo"), "hello")
    }

    func testStripsSpacesAndPunctuation() {
        XCTAssertEqual(Passphrase.normalize("hello world"), "helloworld")
        XCTAssertEqual(Passphrase.normalize("  hello,  world!  "), "helloworld")
        XCTAssertEqual(Passphrase.normalize("don't panic"), "dontpanic")   // apostrophe dropped
        XCTAssertEqual(Passphrase.normalize("a-b_c.d"), "abcd")
    }

    func testCaseAndSpacingVariantsCollapseToOneToken() {
        let variants = ["Foo Bar Baz", "foo bar baz", "FOOBARBAZ", "  foo,bar.baz  ", "Foo-Bar-Baz"]
        let normalized = Set(variants.map(Passphrase.normalize))
        XCTAssertEqual(normalized, ["foobarbaz"])   // every spelling reduces to the same token
    }

    func testKeepsDigits() {
        XCTAssertEqual(Passphrase.normalize("Studio 54"), "studio54")
    }

    func testAllPunctuationYieldsEmpty() {
        XCTAssertEqual(Passphrase.normalize("!!! ,.- "), "")   // nothing to match → simply won't authenticate
    }
}
